common/common-client/src/main/java/org/ow2/proactive/authentication/IAMLoginModule.java

```diff
@@ -0,0 +1,393 @@
+            }
+
+            //Successful login
+            succeeded = true;
+            LOG.debug("authentication succeeded for user '" + authenticator + "'");
+
+            return succeeded;
+
+        } catch (IOException | UnsupportedCallbackException | TicketValidationException e) {
+            LOG.error("", e);
+            throw new LoginException(e.toString());
+        }
+    }
+
+    /**
+     * <p>
+     * This method is called if the LoginContext's overall authentication
+     * succeeded (the relevant REQUIRED, REQUISITE, SUFFICIENT and OPTIONAL
+     * LoginModules succeeded).
+     * <p>
+     *
+     * @exception LoginException
+     *                if the commit fails.
+     *
+     * @return true if this IAMLoginModule's own login and commit attempts
+     *         succeeded, or false otherwise.
+     */
+    @Override
+    public boolean commit() throws LoginException {
+
+        return succeeded;
+    }
+
+    @Override
+    public boolean abort() throws LoginException {
+        boolean result = succeeded;
+        succeeded = false;
+        return result;
+    }
+
+    @Override
+    public boolean logout() throws LoginException {
+        succeeded = false;
+        return true;
+    }
+
+    /**
+     * Creates a ServiceToken given a valid user credentials.
+     * @param username login name of the authenticator.
+     * @param password password of the authenticator.
+     * @return Service token including authentication information.
+     */
+    private String getServiceToken(String username, String password) {
+
+        // Acquire SSO ticket from IAM
+        String ssoTicket = IAMRestClient.getSSOTicket(iamServerUrlPrefix + iamTicketRequest,
+                                                      username,
+                                                      password,
+                                                      ssoTicketHeader);
+        // Acquire JWT based on SSO ticket
+        String serviceToken = IAMRestClient.getServiceToken(iamServerUrlPrefix + iamTicketRequest + "/" + ssoTicket,
+                                                            service);
+
+        CommonUtils.assertNotNull(serviceToken, "no service token produced by IAM");
+
+        return serviceToken;
+
+    }
+
+    /**
+     * Creates a {@link TicketValidator} instance from a class name and map of property name/value pairs.
+     * @param className Fully-qualified name of {@link TicketValidator} concrete class.
+     * @param propertyMap Map of property name/value pairs to set on validator instance.
+     * @return Ticket validator with properties set.
+     */
+    private TicketValidator createTicketValidator(final String className, final Map<String, ?> propertyMap) {
+        CommonUtils.assertTrue(propertyMap.containsKey(IAM_URL_KEY),
+                               "Required property " + IAM_URL_KEY + " not found.");
+
+        final Class<TicketValidator> validatorClass = ReflectUtils.loadClass(className);
```

Comment:
I don't think you need all the `final` keywords in the following.

Reply:
keywords 'final' removed